### ISSUE TITLE
Python Version Mismatch - Manjaro Linux

### ISSUE THREAD

**Issue by pasimko** [JacquesLucke/animation_nodes#815]
```There is a Python version mismatch.

Your Blender build uses: (3, 6, 2)
Animation Nodes has been compiled for: (3, 5, 3)

You have three options:
  1. Try make Blender use another Python version.
     (Blender 2.78/2.79 officially uses Python 3.5.x)
  2. Compile Animation Nodes yourself using the correct Python version.
     (Look in the developer manual for more information)
  3. Create an issue on Github and ask if someone can create a build for you.```

Developer manual is lacking anything on linux, so I figured I'd create an issue here.

**Comment by JacquesLucke** [comment]
Which version of Blender do you use? Blender 2.79 official or a version from build bot?
Sometimes, for some reason, on Linux Blender comes with its own Python version (which is Python 3.5.x in Blender 2.79) and sometimes not. Is it possible for you to somehow get Blender to use Python 3.5?
Otherwise you'll need a Python 3.6.x build of AN for Linux, however right now I don't have one.

**Comment by pasimko** [comment]
I'm using 2.79, installed from the arch repositories, using pacman. After a little searching, I couldn't find anything helpful about changing Python version in Blender. 

I could reinstall from blender.org and see if that works. I will update when I get that done.

**Comment by verbal007** [comment]
I have the same exact issue. Loving Manjaro/Arch Linux... up till now.  :\  *subscribed to thread*

**Comment by jasperges** [comment]
Did you also install Blender with pacman? I'm pretty sure if you install the official Blender 2.79 from [blender.org](https://www.blender.org/download/) your issue will be solved.

**Comment by pasimko** [comment]
Yes, installing from blender.org works. It’s inconvenient, however, because
normal system upgrades/rofi launcher don’t work with it automatically

**Comment by verbal007** [comment]
YES!  Downloading from blender.org worked for me as well. I agree it's slightly inconvenient. Though it's forcing me to run Blender via terminal... which I find is a helpful practice as I receive useful output information. This is a practice I adopted back when I was using Houdini.  Thanks @jasperges .

**Comment by jasperges** [comment]
If you want to build Animation Nodes yourself on Linux, it's fairly straightforward. I just tested this on a fresh install of Manjaro (running in a virtual machine).

```bash
pip install --user cython   # I guess cython is also available with pacman
mkdir ~/src
cd ~/src
git clone https://github.com/JacquesLucke/animation_nodes.git
cd animation_nodes
git checkout tags/v2.0    # For the official 2.0 release
# --noversioncheck so it doesn't check if you're using Python 3.5
# --nocopy so it doesn't try to copy the addon to the Blender addons folder (you can also edit config.py to specify the folder)
# --export so you get a zip archive which you can install in Blender
python setup.py --noversioncheck --nocopy --export
```

But do keep in mind that things you make in this version of Blender and Animation Nodes may be incompatible with the work of others, because of the different Python version.

**Comment by phas0r** [comment]
I had the same problem and I just built it on Arch Linux for Python Version 3.6.3. It worked so far.

Maybe this could be put to the release files for others with Python 3.6.x?

[animation_nodes_v2_0_linux_python3_6_3.zip](https://github.com/JacquesLucke/animation_nodes/files/1486001/animation_nodes_v2_0_linux_python3_6_3.zip)

**Comment by hellocatfood** [comment]
@jasperges  you may also need to install cython3 or cython depending on your version of python:

`sudo apt-get install  cython3`

**Comment by jasperges** [comment]
@hellocatfood That's what `pip install --user cython` is doing. I just prefer to use `pip` whenever possible. On Manjaro (or Arch) the command would be `sudo pacman -S cython` by the way 😸

Manjaro uses Python3.6 by default, but on other Linux distros (and macOS) you might want to use `pip3 install --user cython` (or `pip3.5`, `pip3.6` etc.). I've updated my comment to use `pip3`, to make sure Python3.x is used instead of Python2.x.

**Comment by verbal007** [comment]
Thanks!!

**Comment by dadaromeo** [comment]
@phas0r Thanks for sharing. I reinstalled my system with ubuntu-studio 17.10 and my blender came with Python 3.6.3 which is the default version of Python by the way. So I tried to reinstall AN but couldn't. I searched for the solution and was about to compile blender as a python package in order to recompile AN with Python 3.6.3 but you saved me from the troubles. Thanks again!

**Comment by tariqerwa** [comment]
the 3.6 build by phas0r works well
